ci: run as root within container

rst0git opened the pull request:
When running with default user, it fails with the following error:

```
Run sudo dnf -y install ShellCheck bats golang criu asciidoctor iptables iproute kmod jq bash bash-completion zsh fish
sudo: PAM account management error: Authentication service cannot retrieve authentication info
sudo: a password is required
```

rst0git committed:
ci: run as root within container

When running with default user, it fails with the following error:

sudo: PAM account management error: Authentication service cannot retrieve authentication info
sudo: a password is required

Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>

snprajwal commented:
@rst0git the lint job is failing because of an update to the golangci-lint config spec. `deadline` is now called `timeout` - we would have to make that change in our `.golangci-lint.yml`.

rst0git committed:
golangci: replace deadlie with timeout

golangci-lint fails with the following error:

Error: Failed to run: Error: Command failed: /home/runner/golangci-lint-1.64.6-linux-amd64/golangci-lint config verify
jsonschema: "run" does not validate with "/properties/run/additionalProperties": additional properties 'deadline' not allowed
Failed executing command with error: the configuration contains invalid elements
, Error: Command failed: /home/runner/golangci-lint-1.64.6-linux-amd64/golangci-lint config verify
jsonschema: "run" does not validate with "/properties/run/additionalProperties": additional properties 'deadline' not allowed
Failed executing command with error: the configuration contains invalid elements

    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
Error: Command failed: /home/runner/golangci-lint-1.64.6-linux-amd64/golangci-lint config verify
jsonschema: "run" does not validate with "/properties/run/additionalProperties": additional properties 'deadline' not allowed
Failed executing command with error: the configuration contains invalid elements

Suggested-by: Prajwal S N <prajwalnadig21@gmail.com>
Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>

snprajwal commented:
#158 should fix this, PTAL
